app/utils/misc.tsx

```diff
@@ -6,11 +6,11 @@ import { extendTailwindMerge } from 'tailwind-merge'
 import { extendedTheme } from './extended-theme.ts'
 
 export function getUserImgSrc(imageId?: string | null) {
-	return imageId ? `/resources/user-images/${imageId}` : '/img/user.png'
+	return imageId ? `/resources/images?userImageId=${imageId}` : '/img/user.png'
```

Comment:
This works. It generates this HTML img element src attribute: `/resources/images?src=/resources/images?userImgId=123`. 

- Path `/resources/images` comes from the `OpenImageContextProvider`'s `optimzierEndpoint` prop
- Src query comes from `getUserImgSrc`

Note that we don't need to add `/resources/images` here because we only use the `userImageId` search parameter value in `getImgSources`. Maybe you decided to do this to be more explicit, but just wanted to point it out in case this is an oversight. I also can't think of a better path. 😅

Reply:
Yeah, that’s weird. Any way to make it not that way?

Reply:
The problem is that we have to map from `imageId` to `imageSrc` to signed `imageSrc`. Taking a step back, I think we want to get to the point that adding an image in the UI looks like this:

```tsx
<Img src={user.imgSrc} />
```

I like this a lot because `openimg` becomes just an implementation detail. You can replace `Img` with `img` and it still works (a bit like Remix's `Form` component!) because the `src` is a valid image source. If we want to get to this point, we need to map the `imageId` to `imageSrc` in the loader functions or store the `imageSrc` in the database instead of the `imageId`. What do you think?

Reply:
I prefer the function call. I just want to think about making the path not so odd. I would prefer to make it be like this: `/resources/images/$objectKey`. This way when people remove the notes they don’t have to worry about special handling for notes. So I think we’ll need to do a lookup for the object ID for the UserImage and NoteImages.

Reply:
Okay! The goal is definitely to have `openimg` be configurable in whatever way works best for you:

- The `OpenImageContextProvider` supports a `getSrc` function where you can transform the `Img` props to the img `src` string
- The `getImgResponse` function also has a `getImgParams` function that is called before `getImgSource` to map the `request.url` to `w`, `h`, `fit`, `format`, and `src`.
- We could also refactor `+/resouces/images.tsx` to `+/resources/images/$.tsx` to accept slugs.

So we should have all the flexibility to design a solution that you prefer. I have something ready in a bit!